include/awkward/io/json.h

```diff
@@ -48,16 +48,16 @@ namespace awkward {
       }
     }
 
-    virtual void null() { writer_.Null(); }
-    virtual void boolean(bool x) { writer_.Bool(x); }
-    virtual void integer(int64_t x) { writer_.Int64(x); }
-    virtual void real(double x) { writer_.Double(x); }
-    virtual void string(const char* x, int64_t length) { writer_.String(x, (rj::SizeType)length); }
-    virtual void beginlist() { writer_.StartArray(); }
-    virtual void endlist() { writer_.EndArray(); }
-    virtual void beginrecord() { writer_.StartObject(); }
-    virtual void field(const char* x) { writer_.Key(x); }
-    virtual void endrecord() { writer_.EndObject(); }
+    void null() override { writer_.Null(); }
```

Comment:
It's not visible in this snippet, but this file leaks a dependence on the RapidJSON headers that I'll need to hide in the .cpp file. I think that can be done by defining these `ToJson` classes without implementations in the header and moving all of the implementations to the .cpp file, but that's something that would have to be done manually.